tests/Meziantou.Analyzer.Test/Helpers/ProjectBuilder.Validation.cs

```diff
@@ -453,6 +475,34 @@ private static async Task<string> GetStringFromDocument(Document document)
             root = Formatter.Format(root, Formatter.Annotation, simplifiedDoc.Project.Solution.Workspace);
             return root.GetText().ToString();
         }
+
+        private sealed class CustomDiagnosticProvider : FixAllContext.DiagnosticProvider
+        {
+            private readonly Diagnostic[] _diagnostics;
+
+            public CustomDiagnosticProvider(Diagnostic[] diagnostics)
+            {
+                _diagnostics = diagnostics;
+            }
+
+            public override Task<IEnumerable<Diagnostic>> GetAllDiagnosticsAsync(Project project, CancellationToken cancellationToken)
+            {
+                // TODO I'm not sure of this one
+                return GetProjectDiagnosticsAsync(project, cancellationToken);
+            }
+
+            public override async Task<IEnumerable<Diagnostic>> GetDocumentDiagnosticsAsync(Document document, CancellationToken cancellationToken)
+            {
+                var root = await document.GetSyntaxRootAsync().ConfigureAwait(false);
+                return _diagnostics.Where(d => root == d.Location.SourceTree.GetRoot());
+            }
+
+            public override Task<IEnumerable<Diagnostic>> GetProjectDiagnosticsAsync(Project project, CancellationToken cancellationToken)
+            {
+                // TODO using project.Documents
```

Comment:
While this is not needed in this PR, It would be great to implement it in a similar way as GetDocumentDiagnosticsAsync